### DISCUSSION TITLE
Phone authentication slow?

### DISCUSSION THREAD

**Discussion by dqii** [invertase/react-native-firebase#5949] [Q&A]
https://github.com/invertase/react-native-firebase/issues/5316

This is mostly a follow up on this post. Has anyone else had issues with phone authentication being slow? Specifically, `auth().signInWithPhoneNumber` taking a couple of seconds to complete.

I don't think this is inherent to Firebase, since I've used a couple of iOS apps (from the App Store) that use Firebase Auth, that don't have this same delay.

Has anyone had the issue and solved it? Or have any ideas what might be causing it?

I don't see any comments about it on `firebase-ios-sdk`, only the one issue here, so just throwing this out there.

**Comment by mikehardy** [comment]
Hi there - I get the impression not that many people use phone authentication, or they are very quiet. We don't hear about it very much at all, good or bad. I've used it and did not notice a multi-second delay but perhaps I wasn't paying enough attention. Given what appears to be near total silence on the feature set, if you want to get more details on where the time is going and where it will probably require personal investigation. That said, when doing ios auth with phone vs the emulators in our test e2e harness it is nearly instantaneous. So I don't know if it's the code paths in this module's code

**Comment by uuouter** [comment]
Hey dqii,

Not much help, but just wanted to let you know I haven't encountered this issue at all on iOS.

It works pretty instant and flawless (Dutch number).
